File: tournaments/addBorder/addBorder.cs
string[] addBorder(string[] picture) {
    var rows = picture.Length;
    var cols = picture[0].Length;

    var result = new string[2 + rows];
    result[0] = new String('*', 2 + cols);
    result[1 + rows] = new String('*', 2 + cols);

    for (int i = 0; rows > i; ++i) {
        var newRow = new StringBuilder(cols);
        newRow.Append('*');
        newRow.Append(picture[i]);
        newRow.Append('*');
        result[1 + i] = newRow.ToString();
    }

    return result;
}

File: tournaments/addBorder/addBorder.cpp
typedef std::vector<std::string> Array2D;

Array2D addBorder(Array2D picture) {

  Array2D answer(1, "");

  for (int i = 0; i < picture[0].size() + 2; i++) {
    answer[0] += "*";
  }

  for (int i = 0; i < picture.size(); i++) {
    answer.push_back("*");
    for (int j = 0; j < picture[0].size(); j++) {
      answer[i + 1] += picture[i][j];
    }
    answer[i + 1] += "*";
  }

  answer.push_back(answer[0]);

  return answer;
}
